lib/ts/recipe/oauth2provider/api/utils.ts

```diff
@@ -234,6 +298,23 @@ export async function handleInternalRedirects({
                 redirectTo: authRes.redirectTo,
                 setCookie: mergeSetCookieHeaders(authRes.setCookie, response.setCookie),
             };
+        } else if (response.redirectTo.includes(END_SESSION_PATH)) {
+            response = await recipeImplementation.endSession({
+                params: Object.fromEntries(params.entries()),
+                userContext,
+            });
```

Comment:
The normal auth flow can't get here, so I'm not sure if we should "merge" handling the two different flows here.

Reply:
Separated the redirect functions for logout and login.